### ISSUE TITLE
Form Builder Extra Typeahead updates forms values only after second selection (since 7.1.0)

### ISSUE THREAD

**Issue by glappsi** [danvick/flutter_form_builder#960]
I recently updated the flutter_form_builder package to 7.1.0. Since then the `FormBuilderTypeAhead` components acts weird. form_builder_extra_fields is already at 7.1.0 (and was so before). Everything worked fine. But with the flutter_form_builder upgrade to 7.1.0 the typeahead field doesn't seem to update the forms state on first selection.

Given the following scenario:
```
FormBuilder(
        key: controller.formKey,
        onChanged: controller.onFormChanged
        ...
            FormBuilderTypeAhead(
                name="hallo"
            )
)

controller.onFormChanged = () {
   formKey.currentState!.save();
   print("${formKey.currentState!.value['hallo']}"); // is null on first selection and updated on second (but with first value)
}
```

**Comment by Polgabarro** [comment]
Same bug, currentState shows de UI value, not the real state value. 
For example if I type "a" on the keyboard onChange is triggered and current state only shows "". If I click "save" button now currentState shows the current value.

**Comment by WilliamCunhaCardoso** [comment]
Does it help you?

**Comment by WilliamCunhaCardoso** [comment]
Also, if you get value by name, you do not need to set `onChange`.

In case you want to use `onChange`, you do not need to set up a `FormBuilder`.

**Comment by glappsi** [comment]
@WilliamCunhaCardoso  no this doesn't really help. I need to work with the whole forms value at this point, so using `formKey.currentState!.value`.

**Comment by EslamRabia** [comment]
same problem here

**Comment by Polgabarro** [comment]
¿Any news?

**Comment by danvick** [comment]
Hi @glappsi et al.

Kindly confirm that the issue is fixed in v7.2.1. If that's not the case, feel free to reopen and notify me
